### ISSUE TITLE
Declaration object default type

### ISSUE THREAD

**Issue by kkmuffme** [microsoft/TypeScript#51854]
# Bug Report

Type declaration of object that contains some specific, named keys with a default fallback of other type.

### 🔎 Search Terms

type object default fallback

### 🕗 Version & Regression Information

4.9.3

### ⏯ Playground Link

[Playground link](https://www.typescriptlang.org/play?#code/FDCWDsBcFMCcDMCGBjaACAYgey2g3sGmgM4AO0yo8oyAXGuAK4C2ARnANyFoDaA1tACe9YpFgQA5gF0RYyVwC+wIA)

### 💻 Code

```ts
interface Foo {
  specific: number;
  [key: string]: string;
}
```

### 🙁 Actual behavior

While this type declaration works as expected (e.g. Foo.specific is treated as number, while Foo.random is a string), ts reports an error in the type declaration

### 🙂 Expected behavior

No error reported, since this is valid

**Comment by MartinJohns** [comment]
If you know the types of the specified properties, you can add them to the index signature as I demonstrated using `number`. If you don't know them you have to fall back to `any` or `unknown`.

**Comment by MartinJohns** [comment]
The "random" can also be a key to one of the specific properties, in which case it's not a string. To be safe you to verify the type. Alternatively you can use a type assertion `as string`.

Regardless, the behavior is as intended.

**Comment by MartinJohns** [comment]
> How?

I demonstrated it above. You're declaring a type where **any** `string` key refers to a `string` property, but **also** some specific keys refer to `number` properties. That doesn't work, because these specific keys are part of "any". In order to represent this in a type-safe way you need to be able to say "any string, except these", which is not possible.

> That's fine, I just need some advice on how to declare a type with a known number x of specified keys with type number and an unknown number y of unspecified keys with type string.

You can use an intersection for this. But be aware of the risks I mentioned. It's not type-safe.

```typescript
type Foo = { specific: number } & { [key: string]: string }
```
